### PR TITLE
add custom 404 page for when swagger is enabled

### PR DESCRIPTION
**Pull request by rolfyone**
Add a simple custom 404 page providing a link to swagger, if swagger is enabled.

### CODE REVIEW

File: data/beaconrestapi/src/main/resources/404.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>404 - Page no found</title>
```
Comment (macfarla) [inline data/beaconrestapi/src/main/resources/404.html]:
```suggestion
    <title>404 - Page not found</title>
```

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java [hunk 2]
```diff
@@ -50,14 +58,37 @@
...
     addNetworkHandlers(dataProvider.getNetworkDataProvider());
     addNodeHandlers(dataProvider);
     addValidatorHandlers(dataProvider);
+    addCustomErrorPages(configuration);
+  }
+
+  private void addCustomErrorPages(final ArtemisConfiguration configuration) {
+    if (configuration.getBeaconRestAPIEnableSwagger()) {
+      try {
+        String content = readResource("404.html", Charsets.UTF_8);
```
Comment (macfarla) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java]: nit: should prob pull the filename out as a constant

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java [hunk 3]
```diff
@@ -13,14 +13,22 @@
 
 package tech.pegasys.artemis.beaconrestapi;
 
+import static org.eclipse.jetty.http.HttpStatus.NOT_FOUND_404;
```
Comment (macfarla) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java]: do we need to use jetty 404?

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java]: hrm no that's what it imported, i'll see if there's a better option

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**